src/IndexingMsg.chpl

```diff
@@ -1308,26 +1332,37 @@ module IndexingMsg
             when (DType.BigInt, DType.BigInt) {
                 var x = toSymEntry(gX,bigint);
                 var y = toSymEntry(gY,bigint);
+                var max_size = 1:bigint;
                 if x.max_bits != -1 {
-                    mod(y.a, y.a, x.max_bits);
+                    max_size <<= x.max_bits;
+                    max_size -= 1;
+                    forall y in y.a with (var local_max_size = max_size) {
+                        y &= local_max_size;
+                    }
                 }
                 x.a[slice] = y.a;
              }
             when (DType.BigInt, DType.Int64) {
                 var x = toSymEntry(gX,bigint);
                 var y = toSymEntry(gY,int);
                 var ya = y.a:bigint;
+                var max_size = 1:bigint;
                 if x.max_bits != -1 {
-                    mod(ya, ya, x.max_bits);
+                    max_size <<= x.max_bits;
+                    max_size -= 1;
+                    ya &= max_size;
```

Comment:
you only updated one case of the select statement to do:
```chapel
forall y in y.a with (var local_max_size = max_size) {
    y &= local_max_size;
}
```
You need to update all the others too

Reply:
Updated